solr/core/src/java/org/apache/solr/handler/component/ExpandComponent.java

```diff
@@ -442,23 +442,23 @@ public void modifyRequest(ResponseBuilder rb, SearchComponent who, ShardRequest
 
 
   @Override
-  @SuppressWarnings({"unchecked", "rawtypes"})
   public void handleResponses(ResponseBuilder rb, ShardRequest sreq) {
 
     if (!rb.doExpand) {
       return;
     }
     if ((sreq.purpose & ShardRequest.PURPOSE_GET_FIELDS) != 0) {
       SolrQueryRequest req = rb.req;
-      NamedList expanded = (NamedList) req.getContext().get("expanded");
+      @SuppressWarnings("unchecked")
+      NamedList<Object> expanded = (NamedList<Object>) req.getContext().get("expanded");
       if (expanded == null) {
-        expanded = new SimpleOrderedMap();
+        expanded = new SimpleOrderedMap<>();
         req.getContext().put("expanded", expanded);
       }
 
       for (ShardResponse srsp : sreq.responses) {
-        NamedList response = srsp.getSolrResponse().getResponse();
-        NamedList ex = (NamedList) response.get("expanded");
+        NamedList<Object> response = srsp.getSolrResponse().getResponse();
+        NamedList<?> ex = (NamedList<?>) response.get("expanded");
```

Comment:
`NamedList<?>` here but `NamedList<Object>` above, both for `expanded`, is unexpected at a glance, not yet looked into possible reasons for it.

Reply:
`getResponse()` returns `<Object>` so I went with that, `expanded` has items added to it (L465) so it can't be `<?>`. Meanwhile, we're just reading values out of `ex` so I was lazy and made it wildcard. I suppose we could do `NamedList<SolrDocumentList>` here because we cast it later but I was going for quantity over quality at some point here.